src/historywork/GetHistoryArchiveStateWork.h

```diff
@@ -5,15 +5,15 @@
 #pragma once
 
 #include "history/HistoryArchive.h"
-#include "work/Work.h"
+#include "historywork/DownloadWorkWithMetrics.h"
 
 namespace stellar
 {
 
 class HistoryArchive;
 class GetRemoteFileWork;
 
-class GetHistoryArchiveStateWork : public Work
+class GetHistoryArchiveStateWork : public DownloadWorkWithMetrics
```

Comment:
Sorry, I didn't catch this initially, but it looks like there's an inconsistency here with the `mBytesPerSecond` metric. The issue is the meter assumes that the download finishes whenever the work that inherits `DownloadWorkWithMetrics` finishes. For `GetHistoryArchiveStateWork`, this is fine because the entire class is effectively a wrapper around `GetRemoteFileWork` and does no additional work. However, `GetAndUnzipRemoteFileWork` first downloads the file, then unzips it, then finishes. This means that even if the speed of the download is identical, files downloaded via `GetAndUnzipRemoteFileWork` will always appear to be slower because you count unzip time.

I think the easiest way to fix this is probably just to do your metering in `GetRemoteFileWork`. There's no non-history related work that uses this, so we can probably just do this metering directly in the class. A wrapper class like `DownloadWithMetrics` would be useful if `GetRemoteFileWork` was used for use cases outside of history downloads, but I don't see any. You can mark the `bytesPerSecond` value in the `onSuccess` function still. Since we never retry the work directly, you can mark the retry/failure meter in `onFailureRaise`.

Reply:
Ah right, good catch on the unzip time being included in the denominator. Regarding moving the metrics to `GetRemoteFileWork`, it doesn't make sense to call the metric `history.get.retries` (as its not exactly a retry when incremented from a failing `GetRemoteFileWork`), perhaps `history.get.failures`?